### ISSUE TITLE
Running flows/operations when using multiple directus instances

### ISSUE THREAD

**Issue by MaLub** [directus/directus#19679]
### Describe the Bug

We have observed an odd behaviour when executing flows using multiple directus instances.

We have a flow that iterates over multiple lines and make changes to the collection (change status).
We then have another flow that checks the status of the first flow and makes further changes if necessary (change status).
Both flows were triggered manually.

If both flows run on the same instance, everything works as expected.

If the flows run on different instances, the second flow (the one on the other instance) does not recognise the changes made by the first flow and therefore does not make any changes. (As a workaround, the second flow is repeated by the user until it ends up on the same instance and ultimately leads to success).
All instances have the same redis connected as a cache. The cache has the following cache settings set.

Directus cache settings
```
  - name: CACHE_ENABLED
    value: "true"
  - name: CACHE_STORE
    value: "redis"
  - name: CACHE_REDIS
    value: "redis://redis:6379"
  - name: CACHE_AUTO_PURGE
    value: "true"
  - name: CACHE_AUTO_PURGE_IGNORE_LIST
    value: directus_activity,directus_presets,directus_revisions
  - name: CACHE_REDIS_DB
    value: "4"
  - name: CACHE_NAMESPACE
    value: "directus-cache-prod"
  - name: CACHE_SYSTEM_TTL
    value: "10m"
  - name: CACHE_SCHEMA
    value: "true"
  - name: CACHE_PERMISSIONS
    value: "true"`
```
As a workaround we plan to invalidate the cache at the end of each flow. (https://docs.directus.io/reference/system/utilities.html#clear-the-internal-cache)

Do flow operations invalidate the cache? Or ist a good solution to clear it in every flow (as mentioned above)?
Is this problem potentially solved in a newer version?

Thank you for looking into this!

### To Reproduce

1. run two directus instances in a kubernetes cluster (connected to the same redis)
2. define a first workflow with a simple update operation on a collection
3. define a second workflow with a simple read operation on the same collection 
4. run the first workflow on the first instance
5. run the second workflow on the second instance
6. observe that the second workflow doesn't read the changes made by the first workflow

### Directus Version

v9.24.0

### Hosting Strategy

Self-Hosted (Custom)

**Comment by paescuj** [comment]
This has been fixed in [v10.2.0](https://github.com/directus/directus/releases/tag/v10.2.0) via #18584 👍

**Comment by MaLub** [comment]
@paescuj   Thank you for your quick reply :) Very cool - that helps!
